src/m365/spp/commands/model/model-list.spec.ts

```diff
@@ -0,0 +1,207 @@
+import { sinonUtil } from '../../../../utils/sinonUtil.js';
+import commands from '../../commands.js';
+import command from './model-list.js';
+
+describe(commands.MODEL_LIST, () => {
+  let log: string[];
+  let logger: Logger;
+  let loggerLogSpy: sinon.SinonSpy;
+  let commandInfo: CommandInfo;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').resolves();
+    sinon.stub(telemetry, 'trackEvent').returns();
+    sinon.stub(pid, 'getProcessName').returns('');
+    sinon.stub(session, 'getId').returns('');
+    auth.connection.active = true;
+    commandInfo = cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: async (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: async (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: async (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogSpy = sinon.spy(logger, 'log');
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.get
+    ]);
+  });
+
+  after(() => {
+    sinon.restore();
+    auth.connection.active = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.MODEL_LIST);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('defines correct properties for the default output', () => {
+    assert.deepStrictEqual(command.defaultProperties(), ['AIBuilderHybridModelType', 'ContentTypeName', 'LastTrained', 'UniqueId', 'PublicationType']);
+  });
+
+  it('passes validation when required parameters are valid', async () => {
+    const actual = await command.validate({ options: { siteUrl: 'https://contoso.sharepoint.com/sites/sales' } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('fails validation when siteUrl is not valid', async () => {
+    const actual = await command.validate({ options: { siteUrl: 'invalidUrl' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('correctly handles site is not Content Site', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/sites/portal/_api/web?$select=WebTemplateConfiguration`) {
+        return {
+          WebTemplateConfiguration: 'SITEPAGEPUBLISHING#0'
+        };
+      }
+
+      throw 'Invalid request';
+    });
+
+    await assert.rejects(command.action(logger, { options: { verbose: true, siteUrl: 'https://contoso.sharepoint.com/sites/portal' } }),
+      new CommandError('https://contoso.sharepoint.com/sites/portal is not a content site'));
+  });
+
+
+  it('correctly handles an access denied error', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/sites/portal/_api/web?$select=WebTemplateConfiguration`) {
+        throw 'Attempted to perform an unauthorized operation.';
```

Comment:
It's not realistic that an API throws a plain text error. In reality, the error looks like:

```json
"error": {
    "odata.error": {
      "message": {
        "lang": "en-US",
        "value": "Attempted to perform an unauthorized operation."
      }
    }
  }
```